app/main.py

```diff
@@ -1,6 +1,19 @@
 from typing import Callable
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    results = {}
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs) -> Callable:
+        if args not in results:
+            results[args] = func(*args, **kwargs)
+            print("Calculating new result")
+        elif kwargs and kwargs not in results:
+            results[kwargs] = func(*args, **kwargs)
+            print("Calculating new result")
+        else:
+            print("Getting from cache")
+        return results[args]
+    return wrapper
```

Comment:
```suggestion
        key = args + ... kwargs 
        if key not in results:
            results[key] = func(*args, **kwargs)
            print("Calculating new result")
        else:
            print("Getting from cache")
        return results[key]
    return wrapper
```

Reply:
I can concatenate tuple and tuple, not tuple and dict.
Do not anderstand what is `...` in  `key = args + ... kwargs`.

Reply:
@MaksymProtsak  you have to think about what to use instead of `...` 
to combine `args ` and `kwargs ` into one key

Reply:
key = args + tuple(kwargs.keys())